Tools/AP_Bootloader/can.cpp

```diff
@@ -429,7 +429,7 @@ static void handle_allocation_response(CanardInstance* ins, CanardRxTransfer* tr
         // The allocator has confirmed part of unique ID, switching to the next stage and updating the timeout.
         node_id_allocation_unique_id_offset = msg.unique_id.len;
         send_next_node_id_allocation_request_at_ms -= UAVCAN_PROTOCOL_DYNAMIC_NODE_ID_ALLOCATION_MIN_REQUEST_PERIOD_MS;
-    } else {
+    } else if (msg.node_id != CANARD_BROADCAST_NODE_ID) { // new ID valid? (if not we will time out and start over)
```

Comment:
i don't think this wins anything, just costs flash, the set of bcast is OK, and we stay in DNA

Reply:
I misread the code, currently having `msg.node_id` equal to `CANARD_BROADCAST_NODE_ID` would cause libcanard to execute this line: https://github.com/dronecan/libcanard/blob/6f74bc67656882a4ee51966c7c0022d04fa1a3fb/canard.c#L122

Sure by default the assert doesn't do anything, but it's not really correct to do and the same change was accepted for #28045. We are already careful to not pass that value: https://github.com/ArduPilot/ardupilot/blob/41753b43d740ee64258e754d917cadc83c8d9578/Tools/AP_Bootloader/can.cpp#L828

Reply:
I checked and it does not even consume flash on every periph.